server/data/userClient.js

```diff
@@ -23,29 +23,41 @@ module.exports = {
 
     const { rows } = await db.query(query)
 
-    return rows.map(convertPropertyNames)
+    if (rows) {
+      return rows.map(convertPropertyNames)
+    }
+
+    return []
   },
 
   async getRoUser(nomisId) {
     const query = {
-      text: 'select * from staff_ids where nomis_id = $1',
+      text: 'select * from staff_ids where upper(nomis_id) = upper($1)',
       values: [nomisId],
     }
 
     const { rows } = await db.query(query)
 
-    return convertPropertyNames(rows[0])
+    if (rows && rows[0]) {
+      return convertPropertyNames(rows[0])
+    }
+
+    return undefined
   },
 
   async getRoUserByDeliusId(deliusId) {
     const query = {
-      text: 'select * from staff_ids where staff_id = $1',
+      text: 'select * from staff_ids where upper(staff_id) = upper($1)',
       values: [deliusId],
     }
 
     const { rows } = await db.query(query)
 
-    return convertPropertyNames(rows[0])
+    if (rows && rows[0]) {
+      return convertPropertyNames(rows[0])
+    }
+
+    return undefined
```

Comment:
I have a small problem with returning `undefined` from functions.. I know you've tested it but it makes it hard to know whether there was a `noop` or a misspelled variable ect... what are you thoughts on throwing a custom error?

Reply:
What about null? It's not an error, just a 0 matches. Empty list where the result would be a list, but what's preferred in javascript land for empty single value?

Reply:
Null works.. it's more intentional.. also shows the absence of a value